valerymo/integreatly-operator
# [MGDAPI-2713] feat: add resource requirements for prometheus operator in obserability

# Issue link

https://issues.redhat.com/browse/MGDAPI-2713
# What

Add resource requirements from AMO to OO

# Verification steps



* Checkout this branch
* Update OO index to `quay.io/kevfan/observability-operator-index:v3.0.7` due to requiring new CRD fields that is not in the current index
  * https://github.com/valerymo/integreatly-operator/blob/a6b41a3dc18eb31f8a2a56a70ffc19877aac39a1/products/installation.yaml#L108
* Install RHOAM
```
export INSTALLATION_TYPE=managed-api 
make cluster/prepare/local
make code/run
```
* Once observability stage in complete:
  * Verify resources requirements match - https://issues.redhat.com/browse/MGDAPI-2713?focusedCommentId=19113993&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-19113993
  ```
  oc get alertmanager kafka-alertmanager -n redhat-rhoam-observability -o json| jq '.spec.resources'
  oc get grafana kafka-grafana -n redhat-rhoam-observability -o json| jq '.spec.resources'
  oc get deployment grafana-operator -n redhat-rhoam-observability -o json| jq '.spec.template.spec.containers[0].resources'
  oc get prometheus kafka-prometheus -n redhat-rhoam-observability -o json| jq '.spec.resources'
  oc get deployment prometheus-operator -n redhat-rhoam-observability -o json| jq '.spec.template.spec.containers[0].resources'

  # {
  #   "requests": {
  #     "memory": "200Mi"
  #   }
  # }
  # {
  #   "limits": {
  #     "cpu": "500m",
  #     "memory": "1Gi"
  #   },
  #   "requests": {
  #     "cpu": "100m",
  #     "memory": "256Mi"
  #   }
  # }
  # {}
  # {
  #   "requests": {
  #     "memory": "400Mi"
  #   }
  # }
  # {
  #   "limits": {
  #     "cpu": "200m",
  #     "memory": "400Mi"
  #   },
  #   "requests": {
  #     "cpu": "100m",
  #     "memory": "200Mi"
  #   }
  # }
  ```
